scanner/src/main/kotlin/experimental/NestedProvenanceScanResult.kt

```diff
@@ -167,10 +167,15 @@ data class NestedProvenanceScanResult(
 
         val pathsWithinProvenances = provenances.filter {
             val provenancePath = getPath(it)
-            provenancePath.isEmpty() || path.startsWith("$provenancePath/")
+            provenancePath.isEmpty() || provenancePath == path || provenancePath.startsWith("$path/") || path.startsWith("$provenancePath/")
         }.associateWith { provenance ->
             val provenancePath = getPath(provenance)
-            path.removePrefix("$provenancePath/")
+
+            when {
+                provenancePath.isEmpty() -> path
+                path.startsWith("$provenancePath/") -> path.removePrefix("$provenancePath/")
```

Comment:
Just checking: We don't need the the same logic for `"$path/"`, right?

Reply:
No, above the check is reciprocal to figure out if the subrepository is on the same branch as the provided filter path, but for getting the relative path within the subrepository we don't need it.

Reply:
Thanks for confirming. Would it make sense to put that sentence of yours into a code comment above the `when`?

Reply:
I have added a few explanatory code comments.